### PR TITLE
Provide User Interface to the MythTV's Services

### PR DESCRIPTION
**Pull request by rcrdnalor**
See https://www.mythtv.org/wiki/Services_API .

The class 'MythTVService' can be used in two ways:

Either to be called directly with all parameters to perform an
operation:
Each operation consists of a call to - and a reception of - the
selected MythTVService.
A direct call uses at least the following parameters:
  - service: the service category one is asking for (e.g.: 'Dvr')
    optype = 'POST' # or 'GET'
  - opdata: a dictionary describing the data needed to perform the
            operation.

Or 'MythTVService' is used sequentially by calling:
  - getoperation
  - strip_operation_defaults
  - encode_operation
  - perform_operation

An instance of the 'MythTVService' class can be reused to perform
multiple operations of the same service type.

Example:
  ms = MythTVService('Channel', host)
  op = ms.getoperation('GetChannelInfoList')
  op['opdata']['SourceID'] = 1
  op['opdata']['Details'] = True
  op_stripped = ms.strip_operation_defaults('GetChannelInfoList', op)
  op_encoded = ms.encode_operation(op_stripped)
  if ms.perform_operation(op_encoded):
        # 'ms' is now an object of type 'GetChannelInfoList'
        print(ms.channelinfos[0].chanid)

Notes about naming convention:
The MythTV Python Bindings use lower-case names for their attributes.
The MythTV-Services API use 'CamelCase' notation for the data needed
by 'POST' and returned by 'GET' operations.
During processing of these data, they are converted to lower-case:
Please note, that Python's PEP-08 dislikes 'CamelCase' notation for
naming attributes of a class.

Since every call to 'MythTVService' returns an instance derived from
the base-class 'DictData', one can access these data by either using
av'dictionary-like notation', or using the 'key as attribute'.

TL;DR: Attributes are lower-case names and the following notations
belong to the same object:
    - ms.channelinfo.chanid
    - ms.channelinfo['chanid']
    - ms['channelinfo']['chanid']

A list of operations of the selcted service can be retrieved by
calling 'ms.list_operations()'.
A list of available attributes of an already performed operation
is returned by calling 'ms.list_attributes()'.

Notes about logging:
This module uses the internal logging mechanism from 'logging.py'.
To enable all debug logging to console, start the python script with
$ python3 your_script.py --loglevel debug --verbose all .
For the interactive python shell, replace the scriptname with a
dash ("-").

See 'MythTV/services_api/send.py' for logging of the services_api.

Thank you for contributing to MythTV.

As many of our developers do not visit GitHub it is important to track
all contributions at our central ticket tracker over at code.mythtv.org

To prevent your contributions being overlooked please create a new ticket
there and refer to this pull request. (Bonus points for linking to the
ticket in the pull request, too. It helps us noticed potentially overlooked
pull requests due to missing tickets.)

https://code.mythtv.org/trac/wiki/TicketHowTo
https://code.mythtv.org/trac/newticket

### CODE REVIEW

**Comment by rcrdnalor** [comment]
See ticket #13620 for the basic information and for a simple tutorial.

**Comment by rcrdnalor** [comment]
This PR is to complex for proper future maintenance. PR closed
